bin/migrate-oats-data/applications/decisions/app_modifications/application_modifications_init.py

```diff
@@ -0,0 +1,148 @@
+from common import (
+    OATS_ETL_USER,
+    setup_and_get_logger,
+    add_timezone_and_keep_date_part,
+    AlcsNoiModificationOutcomeCodeEnum,
+)
+from db import inject_conn_pool
+from psycopg2.extras import RealDictCursor, execute_batch
+
+etl_name = "init_application_modifications"
+logger = setup_and_get_logger(etl_name)
+
+
+@inject_conn_pool
+def init_application_modifications(conn=None, batch_size=1000):
+    """
+    This function is responsible for initializing the application_modification in ALCS.
+
+    Args:
+    conn (psycopg2.extensions.connection): PostgreSQL database connection. Provided by the decorator.
+    batch_size (int): The number of items to process at once. Defaults to BATCH_UPLOAD_SIZE.
+    """
+
+    logger.info(f"Start {etl_name}")
+    with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+        with open(
+            "applications/decisions/sql/modifications/application_modification_insert_count.sql",
+            "r",
+            encoding="utf-8",
+        ) as sql_file:
+            count_query = sql_file.read()
+            cursor.execute(count_query)
+            count_total = dict(cursor.fetchone())["count"]
+        logger.info(f"Total Application Modifications data to insert: {count_total}")
+
+        failed_inserts = 0
+        successful_inserts_count = 0
+        last_modification_id = 0
+
+        with open(
+            "applications/decisions/sql/modifications/application_modification_insert.sql",
+            "r",
+            encoding="utf-8",
+        ) as sql_file:
+            query = sql_file.read()
+            while True:
+                cursor.execute(
+                    f"{query} and orr.reconsideration_request_id > {last_modification_id} ORDER BY orr.reconsideration_request_id;"
+                )
+
+                rows = cursor.fetchmany(batch_size)
+
+                if not rows:
+                    break
+                try:
+                    modifications_to_be_inserted_count = len(rows)
+
+                    _insert_application_modifications(conn, batch_size, cursor, rows)
+
+                    successful_inserts_count = (
+                        successful_inserts_count + modifications_to_be_inserted_count
+                    )
+                    last_modification_id = dict(rows[-1])["reconsideration_request_id"]
+
+                    logger.debug(
+                        f"retrieved/inserted items count: {modifications_to_be_inserted_count}; total successfully inserted modifications so far {successful_inserts_count}; last inserted reconsideration_request_id: {last_modification_id}"
+                    )
+                except Exception as err:
+                    logger.exception(err)
+                    conn.rollback()
+                    failed_inserts = count_total - successful_inserts_count
+                    last_modification_id = last_modification_id + 1
```

Comment:
Are rows in ALCS soft deleted or hard deleted? I am wondering if its safe to assume they are always sequential.

Reply:
OATS performs hardDeletes and the PK is a sequential bigint value

Reply:
Wouldn't that mean they are non-sequential, if I originally had IDs 1, 2, 3 and then I delete 2, the rows are now 1, 3?

Reply:
yep, the +1 ensures that we move only 1 step at a time rather than the whole batch( "where" condition in sql). This is useful because of the bad quality of data and is used mostly to identify the failed records in combination with batch_size during development. The resulting ETL has the data issues addressed in the query

Reply:
probably identity + 1 is more accurate than sequential